plugins/topology/src/utils/topology-utils.ts

```diff
@@ -0,0 +1,116 @@
+import { flatten } from 'lodash';
+import { INSTANCE_LABEL } from '../const';
+import { resourceModels } from '../models';
+import {
+  OverviewItem,
+  TopologyDataObject,
+  TopologyDataResources,
+  TopologyResourcesObject,
+} from '../types/topology-types';
+import { K8sResourceKind, WatchK8sResults } from '../types/types';
+
+export const WORKLOAD_TYPES = ['deployments', 'pods'];
+
+export const getTopologyResourceObject = (
+  topologyObject: TopologyDataObject,
+): K8sResourceKind | null => {
+  if (!topologyObject) {
+    return null;
+  }
+  return topologyObject.resource || topologyObject.resources?.obj;
+};
+
+const isGroupVersionKind = (ref: string) => ref.split('~').length === 3;
```

Comment:
How are the `~` delimited reference strings used in topology? This is another OpenShift console-ism that it might be good to remove unless we need it.